doc/stdenv/stdenv.chapter.md

```diff
@@ -116,6 +116,66 @@ On Linux, `stdenv` also includes the `patchelf` utility.
 
 ## Specifying dependencies {#ssec-stdenv-dependencies}
 
+Build systems often require more dependencies than just what `stdenv` provide. This section describes attributes that `stdenv.mkDerivation` accepts and which can be used to make these dependencies available to the build system.
+
+### Simplified overview
+A full reference of the many kinds of dependencies needed to cover all possible cases is provided in the next section, but here is a simplified overview of the most common kinds. It should cover usual cases.
+
+Add to `nativeBuildInputs` dependencies which are executed during the build:
+- those which are needed on `$PATH` during the build, for example `cmake` and `pkg-config`
+- [setup hooks](#ssec-setup-hooks), for example [`makeWrapper`](#fun-makeWrapper)
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for build scripts (with the `--build` flag), which can be the case for e.g. `perl`
+
+Add to `buildInputs` dependencies which will end up copied or linked into the final output, and used at runtime:
+- libraries used by compilers, for example `zlib`,
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for scripts which are installed, which can be the case for e.g. `perl`
+- runtime dependencies added with [`makeWrapper`](#fun-makeWrapper)
```

Comment:
This sounds to me like they should be added, even when it is not the case as you point out later.

Reply:
what does it change? direct string interpolation bypasses splicing, but I don't think merely adding those to buildInputs fixes that if you still use string interpolation?

Reply:
Yeah, I do not think that it would splice the interpolated variables so I am wondering why would one add them to `buildInputs`. I mean the redundancy does not hurt anything but it feels pointless.

Reply:
ah I misunderstood you comment. I was thinking about the design pattern of adding stuff in inputs and the use make wrapper as follows: ` --prefix PYTHONPATH : $PYTHONPATH`

Reply:
removed this line